File: backend/src/main/java/com/example/demo/exception/ErrorCode.java
package com.example.demo.exception;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

@Getter
@RequiredArgsConstructor
public enum ErrorCode {
    // 400 Bad Request
    INVALID_INPUT_VALUE(HttpStatus.BAD_REQUEST, "C001", "Invalid Input Value"),

    // 500 Internal Server Error
    INTERNAL_SERVER_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "C500", "Server Error");

    private final HttpStatus status;
    private final String code;
    private final String message;
}

File: backend/src/main/kotlin/com/example/demo/exception/ErrorCode.kt
package com.example.demo.exception

import org.springframework.http.HttpStatus

enum class ErrorCode(val status: HttpStatus, val code: String, val message: String) {
    // 400 Bad Request
    INVALID_INPUT_VALUE(HttpStatus.BAD_REQUEST, "C001", "Invalid Input Value"),

    // 409 Conflict
    DUPLICATE_REQUEST(HttpStatus.CONFLICT, "C409", "Duplicate Request Detected"),

    // 500 Internal Server Error
    INTERNAL_SERVER_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "C500", "Server Error")
}
